File: s3PythonSDK/sample/WebsiteSerialTesting.py
from botocore.exceptions import ClientError
from client import client


def main(arg):
    try:
        client.create_bucket(
            CreateBucketConfiguration={'LocationConstraint': 'ap-northeast-1'},
            Bucket=arg[0],
        )
        client.put_object(
            ACL='public-read',
            Body='404testchttl<br><title>chttl</title>',
            Bucket=arg[0],
            Key='error.html',
        )
        client.put_object(
            ACL='public-read',
            Body='<title>chttl</title>Hello World!',
            Bucket=arg[0],
            Key='index.html',
        )
        #print("Put Website Configurations..")
        client.put_bucket_website(
            Bucket=arg[0],
            WebsiteConfiguration={
                'ErrorDocument': {
                    'Key': 'error.html'
                },
                'IndexDocument': {
                    'Suffix': 'index.html'
                },

            },
        )
        #print("Get Website Configurations..")
        response = client.get_bucket_website(
            Bucket=arg[0],
        )
        #print('IndexDocument : ' + repr (response['IndexDocument']))
        #print('ErrorDocument : ' + repr (response['ErrorDocument']))
        #print("Delete Website Configurations..\n")
        client.delete_bucket_website(
            Bucket=arg[0],
        )
        client.put_bucket_website(
            Bucket=arg[0],
            WebsiteConfiguration={
                'IndexDocument': {
                    'Suffix': 'index.html'
                },
                'RoutingRules': [
                    {
                        'Redirect': {
                            'HostName': 'www.google.com',
                            'Protocol': 'https',
                        }
                    },
                ]
            },
        )
        #print("Put Website Configurations with redirect all request")
        #print("Get Website Configurations..")
        response = client.get_bucket_website(
            Bucket=arg[0],
        )
        #print('IndexDocument : ' + repr (response['IndexDocument']))
        #print('RoutingRules : ' + repr (response['RoutingRules']))
        #print("Delete Website Configurations..\n")
        client.delete_bucket_website(
            Bucket=arg[0],
        )

        #print("\nClean up..")
        result = client.list_objects_v2(
            Bucket=arg[0],
        )
        if 'Contents' in result:
            for r in result['Contents']:
                client.delete_object(
                    Bucket=arg[0],
                    Key=r['Key']
                )
        client.delete_bucket(
            Bucket=arg[0],
        )
        print("Website Serial Test Done ! \n")
    except ClientError as e:
        print("Error operation : " + e.operation_name)
        print("Error response : " + e.response['Error']['Message'])

File: s3PhpSDK/sample/WebsiteSerialTesting.php
<?php
/*
 * test 1. Basic putBucket
 * 		2. put Object
 * 		3. put BucketWebsite (Basic & with redirect rules)
 * 		4. get BucketWebsite
 * 		5. Delete BucketWebsite
 */

use Aws\S3\Exception\S3Exception;
require 'client.php';
$bucketname=$argv[1];

try{
	echo "Website Serial Testing...\n";
	$client->createBucket(array(
		'Bucket' => $bucketname,
		'ACL' 	 => 'public-read'	
	));
	
	$client->putObject(array(
			'Body'   => '404testchttl<br><title>chttl</title>',
			'Bucket' => $bucketname,
			'Key'	 => 'error.html',
			'ACL' 	 => 'public-read',
			'WebsiteRedirectLocation' => 'http://google.com'
	));
	
	$client->putObject(array(
			'Body'   => 'Hello world!',
			'Bucket' => $bucketname,
			'Key'	 => 'index.html',
			'ACL' 	 => 'public-read'
	));
	$client->putBucketWebsite(array(
			'Bucket' => $bucketname,
			'WebsiteConfiguration' => [
				'ErrorDocument' => array(
					'Key' => 'error.html'
					),
				'IndexDocument' => array(
					'Suffix' => 'index.html'
					),
			]
	));
	
	$result=$client->getBucketWebsite( array(
			'Bucket' => $bucketname
	));
	
	//apply routing rules, may take some time for applying new rules  
	$client->putBucketWebsite(array(
			'Bucket' => $bucketname,
			'WebsiteConfiguration' => [
				'ErrorDocument' => array(
						'Key' => 'error.html'
				),
				'IndexDocument' => array(
						'Suffix' => 'index.html'
				),
				'RoutingRules' => array(
						'RoutingRule' =>array(
								'Redirect' =>array(
										'HostName' => 'www.google.com'
								),
						),
				),
			]
	));
	
	$result=$client->getBucketWebsite( array(
			'Bucket' => $bucketname
	));
	
	$client->deleteBucketWebsite(array(
			'Bucket' => $bucketname
	));
	
	$client->deleteObject(array(
			'Bucket' => $bucketname,
			'Key'	 => 'error.html'
	));
	$client->deleteObject(array(
			'Bucket' => $bucketname,
			'Key'	 => 'index.html'
	));
	
	$client->deleteBucket(array(
			'Bucket' => $bucketname
	));
}catch (S3Exception $e) {
    echo "Caught an AmazonServiceException.", "\n";
    echo "Error Message:    " . $e->getAWSErrorMessage(). "\n";
    echo "HTTP Status Code: " . $e->getStatusCode(). "\n";
    echo "AWS Error Code:   " . $e->getAwsErrorCode(). "\n";
    echo "Error Type:       " . $e->getAwsErrorType(). "\n";
    echo "Request ID:       " . $e->getAwsRequestId(). "\n";
}
